server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/ScanDataSource.java

```diff
@@ -234,32 +239,60 @@ private SortedKeyValueIterator<Key,Value> createIterator() throws IOException {
     }
   }
 
-  @Override
-  public void close(boolean sawErrors) {
-
+  private void returnIterators() {
     if (memIters != null) {
-      tablet.returnMemIterators(memIters);
-      memIters = null;
-      tablet.returnFilesForScan(fileReservationId);
-      fileReservationId = -1;
-    }
-
-    synchronized (tablet) {
-      if (tablet.removeScan(this) == 0) {
-        tablet.notifyAll();
+      log.trace("Returning mem iterators for {}, scanId:{}, fid:{}", tablet.getExtent(),
+          scanDataSourceId, fileReservationId);
+      try {
+        tablet.returnMemIterators(memIters);
+      } finally {
+        memIters = null;
+        log.trace("Returning file iterators for {}, scanId:{}, fid:{}", tablet.getExtent(),
+            scanDataSourceId, fileReservationId);
+        try {
+          tablet.returnFilesForScan(fileReservationId);
+        } catch (IllegalArgumentException | IllegalStateException e) {
+          if (log.isWarnEnabled()) {
+            log.warn("Error Returning file iterators for scan: {}, :{}", scanDataSourceId,
+                e.getMessage());
+            log.warn("{}", ExceptionUtils.getStackTrace(e));
+          }
```

Comment:
Use of ExceptionUtils is not needed here at all. If we want the stack trace, that can be done as a built-in feature of the logging framework. We also don't need the check if the warning level is enabled. Just make the exception the last parameter after all the formatString's parameters, and the logging framework will automatically include the exception's message and stack trace.

```suggestion
          log.warn("Error returning file iterators for scan: {}", scanDataSourceId, e);
```